### ISSUE TITLE
[Feature] Use of MetricCollection in LightningModule

### ISSUE THREAD

**Issue by tourniert** [meteofrance/mfai#48]
## **Overview**
Change the implementation of metrics in the SegmentationLightningModule by using a MetricCollection (see https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metriccollection).

- Make use of existing `pytorch` classes
- Will remove some the "for" loop in train, valid and test forwards -> the update of all metrics took one line : `self.valid_metrics.update(logits, y)`
- More pythonic
